src/vmm/src/lib.rs

```diff
@@ -379,11 +381,33 @@ impl VMM {
     // Create and add a serial console to the VMM.
     fn add_serial_console(&mut self) -> Result<()> {
         // Create the serial console.
+        let serial;
         let interrupt_evt = EventFd::new(libc::EFD_NONBLOCK).map_err(Error::IO)?;
-        let serial = Arc::new(Mutex::new(SerialWrapper(Serial::new(
-            interrupt_evt.try_clone().map_err(Error::IO)?,
-            stdout(),
-        ))));
+        match &self.kernel_cfg.serial_console_path {
+            Some(x) => {
+                let input_pipe_string = x.to_owned() + "_in";
+                let output_pipe_string = x.to_owned() + "_out";
+                let input_pipe_path = PathBuf::from(input_pipe_string);
+                let output_pipe_path = PathBuf::from(output_pipe_string);
```

Comment:
This looks like it would be easier if we just let the user specify both paths, instead of creating files starting from a prefix, which some users could find confusing. I also think it'd be easier to work with `&str`s or `P: AsRef<Path>` instead of owned `String`s and `PathBuf`s to make use of borrowing wherever possible.